src/components/TaskRunArtifacts.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <div class="task-run-artifacts">
+    <div class="task-run-artifacts__button-group-container">
+      <slot name="actions" />
+      <ViewModeButtonGroup class="task-run-artifacts__view-mode-button-group" />
+    </div>
+
+    <RowGridLayoutList :items="artifacts">
+      <template #default="{ item }">
+        <router-link :to="routes.artifact(item.id)">
+          <ArtifactCard :artifact="item" :condense="condense" class="task-run-artifacts__artifact" />
+        </router-link>
+      </template>
+
+      <template v-if="artifactsSubscription.executed" #empty>
+        <p-empty-state>
+          <template #description>
+            <p-markdown-renderer :text="emptyMessage" />
+          </template>
+        </p-empty-state>
+      </template>
+    </RowGridLayoutList>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import ArtifactCard from '@/components/ArtifactCard.vue'
+  import RowGridLayoutList from '@/components/RowGridLayoutList.vue'
+  import ViewModeButtonGroup from '@/components/ViewModeButtonGroup.vue'
+  import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
+  import { localization } from '@/localization'
+  import { TaskRun, isTerminalStateType } from '@/models'
+  import { ArtifactsFilter } from '@/models/Filters'
+  import { activeViewMode } from '@/utilities/activeViewMode'
+
+  const props = defineProps<{
+    taskRun: TaskRun,
+  }>()
+
+  const condense = computed(() => activeViewMode.value !== 'grid')
+
+  const api = useWorkspaceApi()
+  const routes = useWorkspaceRoutes()
+
+  const artifactsFilter = computed<ArtifactsFilter>(() => {
+    return {
+      artifacts: {
+        taskRunId: [props.taskRun.id],
+        notType: ['result'],
+      },
+    }
+  })
+  const artifactsSubscription = useSubscription(api.artifacts.getArtifacts, [artifactsFilter], { interval: 10000 })
```

Comment:
Can we only add the interval if it's running?

Reply:
I have a ticket open for that, going to tackle separately

Reply:
https://github.com/PrefectHQ/prefect-ui-library/issues/1320